java-app-backend/ListFacesFunction/src/main/java/com/webapp/listfaces/ListFacesHandler.java

```
@@ -0,0 +1,59 @@
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+
+public class ListFacesHandler implements RequestHandler<Object, APIGatewayProxyResponseEvent> {
+
+    private static final String TABLE_NAME = System.getenv("TableName");
+    private static final String CF_DISTRIBUTION = System.getenv("CloudFrontDistribution");
+    private static final String PLACEHOLDER_URL = "https://" + CF_DISTRIBUTION + "/index/static/aws_logo.png" ;
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequest(Object input, Context context) {
+
+        final LambdaLogger logger = context.getLogger();
+
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Access-Control-Allow-Origin", "*");
+        headers.put("Access-Control-Allow-Methods", "POST,GET");
+        APIGatewayProxyResponseEvent apiGatewayProxyResponseEvent = new APIGatewayProxyResponseEvent()
+                .withHeaders(headers);
+
+        List<FacePojo> list = new ArrayList<FacePojo>();
+        logger.log("DDB Table: " + TABLE_NAME);
+        AmazonDynamoDB client = AmazonDynamoDBClientBuilder.standard().build();
+        ScanRequest scanRequest = new ScanRequest().withTableName(TABLE_NAME);
+        ScanResult result = client.scan(scanRequest);
+
+        for (Map<String, AttributeValue> item : result.getItems()) {
+            String fullName = item.get("FullName").getS();
+            String imageUrl = "https://" + CF_DISTRIBUTION + "/" + item.get("BucketKey").getS();
+            list.add(new FacePojo(imageUrl, fullName));
+        }
+        try {
+            return apiGatewayProxyResponseEvent
+                    .withStatusCode(200)
+                    .withBody(OBJECT_MAPPER.writeValueAsString(list));
+        } catch (JsonProcessingException e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)